### PR TITLE
Fix issues #131 #132 #136 for coadd reader

### LINKED ISSUE #131: coadd reader: native quantities with `slot_PsfFlux` and `ext_shapeHSM` are not per band.  Change to `base_PsfFlux` and explicitly use and document `base_SdssShape``?
In [Lines 145-147 of the coadd reader](https://github.com/LSSTDESC/gcr-catalogs/blob/master/GCRCatalogs/dc2_coadd.py#L145), we have:
```python
    modifiers['psFlux_{}_lsst'.format(band)] = '{}_slot_PsfFlux_flux'.format(band)
    modifiers['psFlux_flag_{}_lsst'.format(band)] = '{}_slot_PsfFlux_flag'.format(band)
    modifiers['psFlux_err_{}_lsst'.format(band)] = '{}_slot_PsfFlux_fluxSigma'.format(band)
```

But looking at the native quantities, I only find:
``` python 
[
    ...
    'slot_PsfFlux_flag',
    'slot_PsfFlux_flux',
    'slot_PsfFlux_fluxSigma',
    ...
]
```
and 
```python
[
    ...
    'u_base_PsfFlux_flag',
    'u_base_PsfFlux_flux',
    'u_base_PsfFlux_fluxSigma',
    ...
]
```
So this is probably a typo?

### LINKED ISSUE #132: drop `_lsst` in coadd reader; add `mag_<band>` to extragalactic catalog readers
We have been using `mag_<band>_<filter>` in the extragalactic catalog readers as there are `mag_g_lsst` and `mag_g_sdss` etc. For DC2 catalogs, the filter postfix seems unnecessary, so we will drop it from the coadd reader. And for extragalactic catalog readers, `mag_<band>` will be an alias of `mag_<band>_lsst`.

We will modify the schema and the readers accordingly.

### LINKED ISSUE #136: Should we have both 'good' and 'clean' in DC2 coadd reader?
In the DC2 coadd reader, `good` is defined as both no flagged pixels and not deblended (`deblend_skipped`).  (code snippet below).

Should we instead define `good` as no flagged pixels, and a new column `clean` = `good`& `deblend_skipped`?

```
modifiers['good'] = (
            create_basic_flag_mask,
            'deblend_skipped',
            'base_PixelFlags_flag_edge',
            'base_PixelFlags_flag_interpolatedCenter',
            'base_PixelFlags_flag_saturatedCenter',
            'base_PixelFlags_flag_crCenter',
            'base_PixelFlags_flag_bad',
            'base_PixelFlags_flag_suspectCenter',
            'base_PixelFlags_flag_clipped',
        )
```

### PR DESCRIPTION
**Pull request by djperrefort**
Addresses Issue #131 and fixes the mapping of band-specific PSF values to use `_base_` instead of `_slot_` values.

[edited by @yymao: fix #131, fix #132, fix #136]

### CODE REVIEW

**Comment by yymao** [comment]
Thanks @djperrefort. Can you also address part of #132 and drop the `_lsst` postfix in the coadd reader? I'll work on the extragalactic catalog readers.

**Comment by yymao** [comment]
Can (or should) we change `SNR` and `CModel` to lower cases? That seems to be more consistent with other quantities that are not in DPDD. @djperrefort @wmwv ?

**Comment by djperrefort** [comment]
@yymao, @wmwv Something along the lines of `cModel` might be more consistent.

**Comment by yymao** [comment]
@djperrefort also need to drop `_lsst` in `SCHEMA.md`. 

@wmwv earlier you mentioned something not quite right about `centroidX`?

File: GCRCatalogs/SCHEMA.md [hunk 1]
```diff
@@ -106,20 +106,20 @@ Quantity Label | Unit | Definition
...
+`Ixy_<band>` | asec2 | Adaptive second moment of the source intensity in `<band>`. 
+`IxxPSF_<band>` | asec2 | Adaptive second moment for the PSF  in `<band>`. 
+`IyyPSF_<band>` | asec2 | Adaptive second moment for the PSF  in `<band>`. 
+`IxyPSF_<band>` | asec2 | Adaptive second moment for the PSF  in `<band>`. 
+`I_flag_<band>` | - | Flag for issues with `Ixx_<band>`, `Ixx_<band>`, and `Ixx_<band>.` 
 `mag_<band>_CModel` | mag | Apparent magnitude in `<band>`, fitted by CModel.
 `magerr_<band>_CModel` | mag | Error value for `mag_<band>_cmodel.` 
 `SNR_<band>_CModel` | - | Signal to noise ratio for magnitude in `<band>`, fitted by CModel.
-` psf_fwhm_<band>` | pixels | The full width at half maximum of the PSF 
+`psf_fwhm_<band>` | pixels | The full width at half maximum of the PSF
```
Comment (wmwv) [inline GCRCatalogs/SCHEMA.md]: "PSF FWHM calculated from 'base_SdssShape'"

The audience here knows what FWHM means.  It's what it was calculated from that is the question.

Units are arcseconds.  (Follow through the units of the calculation).

File: GCRCatalogs/SCHEMA.md [hunk 2]
```diff
@@ -97,7 +97,7 @@ The schema for DC2 Coadd Catalogs follow the following rules:
 - For quantities that are defined in [LSST DPDD](https://lse-163.lsst.io/dpdd.pdf), we follow DPDD's naming scheme.
 - For quantities that are defined the above "Schema for Extragalatic Catalogs", we follow Extragalatic Catalogs' naming scheme.
 - For quantities that are defined in both, we provide aliases so both naming schemes would work.
-- For quantities that are defined in neither, we define them below:
+- For quantities that are defined in neither, we define them below. :
```
Comment (wmwv) [inline GCRCatalogs/SCHEMA.md]: This statement is confusing and not entirely accurate.  We should list all of the things we define here and state whether or not they're from the DPDD.

Comment (yymao) [reply GCRCatalogs/SCHEMA.md]: this one addressed.

File: GCRCatalogs/SCHEMA.md [hunk 3]
```diff
@@ -106,20 +108,20 @@ Quantity Label | Unit | Definition
...
+`Ixx_<band>` | asec2 | Adaptive second moment of the source intensity in `<band>`.
+`Iyy_<band>` | asec2 | Adaptive second moment of the source intensity in `<band>`. 
+`Ixy_<band>` | asec2 | Adaptive second moment of the source intensity in `<band>`. 
+`IxxPSF_<band>` | asec2 | Adaptive second moment for the PSF  in `<band>`. 
+`IyyPSF_<band>` | asec2 | Adaptive second moment for the PSF  in `<band>`. 
+`IxyPSF_<band>` | asec2 | Adaptive second moment for the PSF  in `<band>`. 
+`I_flag_<band>` | - | Flag for issues with `Ixx_<band>`, `Ixx_<band>`, and `Ixx_<band>.` 
 `mag_<band>_CModel` | mag | Apparent magnitude in `<band>`, fitted by CModel.
 `magerr_<band>_CModel` | mag | Error value for `mag_<band>_cmodel.` 
 `SNR_<band>_CModel` | - | Signal to noise ratio for magnitude in `<band>`, fitted by CModel.
```
Comment (yymao) [inline GCRCatalogs/SCHEMA.md]: change to lower case `snr`

File: GCRCatalogs/SCHEMA.md [hunk 4]
```diff
@@ -106,20 +108,20 @@ Quantity Label | Unit | Definition
 `centroidX_err` | pixels | Error value for `centroidX`.
```
Comment (yymao) [inline GCRCatalogs/SCHEMA.md]: change `centroidX` to `x` and `centroidY` to `y`, and also for error and flag.

File: GCRCatalogs/SCHEMA.md [hunk 5]
```diff
@@ -106,20 +108,20 @@ Quantity Label | Unit | Definition
...
+`psFluxErr_<band>` | nmgy | Error value for `psFlux_<band>`. 
+`psFlux_flag_<band>` | - | Flag for issues with `psFlux_<band>`. 
+`Ixx_<band>` | asec2 | Adaptive second moment of the source intensity in `<band>`.
+`Iyy_<band>` | asec2 | Adaptive second moment of the source intensity in `<band>`. 
+`Ixy_<band>` | asec2 | Adaptive second moment of the source intensity in `<band>`. 
+`IxxPSF_<band>` | asec2 | Adaptive second moment for the PSF  in `<band>`. 
+`IyyPSF_<band>` | asec2 | Adaptive second moment for the PSF  in `<band>`. 
+`IxyPSF_<band>` | asec2 | Adaptive second moment for the PSF  in `<band>`. 
+`I_flag_<band>` | - | Flag for issues with `Ixx_<band>`, `Ixx_<band>`, and `Ixx_<band>.` 
 `mag_<band>_CModel` | mag | Apparent magnitude in `<band>`, fitted by CModel.
```
Comment (yymao) [inline GCRCatalogs/SCHEMA.md]: I vote for `cmodel`. @wmwv ?

File: GCRCatalogs/SCHEMA.md [hunk 6]
```diff
@@ -106,20 +108,20 @@ Quantity Label | Unit | Definition
...
 `magerr_<band>_CModel` | mag | Error value for `mag_<band>_cmodel.` 
 `SNR_<band>_CModel` | - | Signal to noise ratio for magnitude in `<band>`, fitted by CModel.
-` psf_fwhm_<band>` | pixels | The full width at half maximum of the PSF 
+`psf_fwhm_<band>` | pixels | The full width at half maximum of the PSF 
 `good` | - | Whether the source contains any corrupted pixels. 
 `I_flag` | - | Flag for issues with `Ixx`, `Iyy`, and `Ixy`. 
 `blendedness` | - | measure of how flux is affected by neighbors: (1 - flux.child/flux.parent) (see 4.9.11 of [1705.06766](https://arxiv.org/abs/1705.06766))
```
Comment (yymao) [inline GCRCatalogs/SCHEMA.md]: Add also `extendedness`.

**Comment by yymao** [comment]
@wmwv maybe change `*` to another symbol? `*` is used by markdown to do emphasis.

**Comment by EiffL** [comment]
@djperrefort Just realized that the pixel scale is  missing from the configuration file, I'm adding it to this branch

**Comment by yymao** [comment]
Kudos to @djperrefort for noticing @wmwv's subtle difference between `*` and `x`.

**Comment by djperrefort** [comment]
Other than the undecided name for `CModel` are there any other changes that need to be made for this PR?

**Comment by yymao** [comment]
Looks good to me. But I think we should merge it on Monday, not over the weekend.

@wmwv, choose among `CModel`, `cModel`, and `cmodel`?

**Comment by djperrefort** [comment]
@yymao Agreed.

**Comment by EiffL** [comment]
:-/ it's going to be annoying if we work on notebooks over the weekend and
the name change ...

**Comment by yymao** [comment]
@EiffL, how about you make a decision and we fellow you?

**Comment by EiffL** [comment]
Concerning the cmodel stuff ? I think cmodel or cModel are fine. Consider
this: http://www.sdss.org/dr12/algorithms/magnitudes/#cmodel
They use cModel in the text

**Comment by yymao** [comment]
`cModel` it is!

**Comment by yymao** [comment]
Now also addresses #136.

File: GCRCatalogs/dc2_coadd.py [hunk 7]
```diff
@@ -133,38 +131,45 @@ def _generate_modifiers(pixel_scale=0.2):
...
+        modifiers['good'] = (create_basic_flag_mask,) + not_good_flags
+
+        modifiers['clean'] = (
+            create_basic_flag_mask,
+            'deblend_skipped',
+        ) + not_good_flags
+
         # cross-band average, second moment values
-        modifiers['I_flag'] = 'slot_Shape_flag'
+        modifiers['I_flag'] = 'ext_shapeHSM_HsmSourceMoments'
```
Comment (yymao) [inline GCRCatalogs/dc2_coadd.py]: @wmwv `ext_shapeHSM_HsmSourceMoments` does not exist in the hdf5 files.

Comment (yymao) [reply GCRCatalogs/dc2_coadd.py]: maybe you mean `ext_shapeHSM_HsmSourceMoments_flag`?

Comment (wmwv) [reply GCRCatalogs/dc2_coadd.py]: Yes, thank you.

Comment (yymao) [reply GCRCatalogs/dc2_coadd.py]: OK, fixed.

**Comment by yymao** [comment]
@djperrefort, I have been several commits since the last time you committed. Do you mind taking another look to see if everything makes sense before I merge this? Thanks!

**Comment by djperrefort** [comment]
@yymao It all looks good to me!

**Pull request merged**